### ISSUE TITLE
Maybe searches in urlbar should respect the browser.search.openintab

### ISSUE THREAD

**Issue by yfdyh000** [Quicksaver/The-Fox--Only-Better#153]
If the pref-showOnlyNonEmptySearchBar is enabled. Or, add a option?

**Comment by Quicksaver** [comment]
When the search comes from the location bar, it should still obey location bar rules, that preference is for the search bar, not for searches in general (for instance you can search from the location bar even without the add-on, and it still doesn't obey pref as expected). I'd rather leave the location bar's behavior as-is.

You can use an add-on like Tab Mix Plus to always open new tabs from the location bar if you want. You can also ctrl+left click or middle click the go button or a suggestion in the panel, or press alt+enter; all of those will open in a new tab by default.

**Comment by yfdyh000** [comment]
Yes, I understand the reason. But, I hope opens urls in the current page and open a new tab for search results always.

Can you add an option or pref to respect it for urlbar searches always when search bar has disappear. Either, respect it if the focus source is Ctrl+K and things like that, although it may be confusing.

**Comment by Quicksaver** [comment]
There are two main reasons I feel strongly about not doing this:
- [by definition, that preference acts on searches from the search bar specifically](http://kb.mozillazine.org/About:config_entries). I agree that this is probably an outdated description, from a time when you couldn't search from the location bar. But it still stands that obeying that preference can possibly lead to some confusion in interpreting its results in the location bar.
- I could add an entire new preference specific to this add-on to control this, but this part is buried deep within the location bar's native code, which is heavily modified by many extensions like Tab Mix Plus, Tab Utilities, and other similar ones. It's a trivial change, sure, but it's in a place of the code that when changed would break those extensions very easily, and often. This happens when minor changes are done to Firefox, and the more add-ons that change/rely on those parts of the code the worse it becomes. I'd rather not pile on.

> respect it if the focus source is Ctrl+K and things like that

Definitely not, that's even worse. Different outcomes should come either from different places or from different actions, not from how you got to those places; i.e. typing in the location bar is typing in the location bar, it doesn't matter how you started typing there.

If I were to add this option, it would respect all search actions from the location bar, regardless of any other preferences, of how the location bar was focused, etc.

**Comment by yfdyh000** [comment]
Um, it seems there is no way to my needs, searches open in a new tab always and open urls is not affected?

**Comment by Quicksaver** [comment]
Not unless I can find a "safe" way to do it. I'll check this out again either later or tomorrow, but I really can't make any promises.

**Comment by Quicksaver** [comment]
I'm reopening to remind me to check it out again at a later time. I'll still try to do something about it eventually, but I have to put it in the backburner at least for a while, I hope you understand.

**Comment by yfdyh000** [comment]
I understand. Thanks for your efforts for a better future.
